### DISCUSSION TITLE
setting input file subblocks as active/inactive

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#23299] [Q&A General] [answered]
I have an input file with this format:

```
[BCs]
  active = ''
  [fixed_x]
    type = DirichletBC
    ....
  []
  [InclinedNoDisplacementBC]
    [inclined]
      boundary = inclined_nodes
      ...
    []
  []
[]
```

I want to switch on my BCs from the command line, but it doesn't seem I can set the inclined BC as active using the top-level parameters? I've tried appending both `BCs/active="fixed_x InclinedNoDisplacementBC/inclined"` and `BCs/active="fixed_x" BCs/InclinedNoDisplacementBC/active="inclined"` to the command line but neither seem to work.

**Comment by GiudGiud** [comment, marked as answer]
Hello

InclineNoDisplacementBC is an action within the BCs block. I m afraid it won't work the same way as two regular BCs and you pick which is active.

But, if this particular action supports it, you should be able to do:
```
[BCs]
  inactive / active = 'fixed_x InclinedNoDisplacementBC'
  [fixed_x]
    type = DirichletBC
    ....
  []
  [InclinedNoDisplacementBC]
    ~~inactive/active = 'inclined'~~
    [inclined]
      boundary = inclined_nodes
      ...
    []
  []
[]
```

but you d have to look at it closely to know if that action supports active/inactive

**Reply by jessecarterMOOSE** [reply]
It looks like I can turn the whole action on (or off) with `BCs/active=InclinedNoDisplacementBC`, which works for now.

**Reply by GiudGiud** [reply]
ok then no issue?

**Reply by GiudGiud** [reply]
or do you need to be able to specify within this nested action which BCs should be active?

**Reply by jessecarterMOOSE** [reply]
Yeah right now I either have inclined BCs or I don't, so turning them all off is ok.

**Reply by jessecarterMOOSE** [reply]
> or do you need to be able to specify within this nested action which BCs should be active?

Maybe someday!
